openmrs-19/src/main/java/org/motechproject/openmrs19/tasks/OpenMRSTaskDataProvider.java

```diff
@@ -0,0 +1,86 @@
+package org.motechproject.openmrs19.tasks;
+
+import org.motechproject.commons.api.AbstractDataProvider;
+import org.motechproject.openmrs19.domain.OpenMRSProvider;
+import org.motechproject.openmrs19.service.OpenMRSProviderService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.motechproject.openmrs19.tasks.OpenMRSTasksConstants.BY_UUID;
+import static org.motechproject.openmrs19.tasks.OpenMRSTasksConstants.NAME;
+import static org.motechproject.openmrs19.tasks.OpenMRSTasksConstants.PACKAGE_ROOT;
+import static org.motechproject.openmrs19.tasks.OpenMRSTasksConstants.PROVIDER;
+import static org.motechproject.openmrs19.tasks.OpenMRSTasksConstants.UUID;
+
+/**
+ * This is the OpenMRS task data provider that is registered with the task module as a data source.
+ * It allows retrieving objects from OpenMRS model and using it in tasks.
+ */
+@Service("openMRSTaskDataProvider")
+public class OpenMRSTaskDataProvider extends AbstractDataProvider {
+    private static final Logger LOGGER = LoggerFactory.getLogger(OpenMRSTaskDataProvider.class);
+    private static final List<Class<?>> SUPPORTED_CLASS = Arrays.asList(OpenMRSProvider.class);
```

Comment:
Yes, as I wrote, there will be more supported classes soon (I have to change name of variable to SUPPORTED_CLASSES). Now there is only Provider, because I have still some questions about Encounter and Patient.